### PR TITLE
HL-5

### PR DESCRIPTION
**Pull request by OptimusOpus**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-5)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -55,6 +126,10 @@ const getProvider = () => {
   return provider;
 };
 
+const getCurrentBlockHeight = async provider => {
```
Comment (brettvniekerk) [inline src/index.js:129]: this could be an inferred return

File: src/index.js [hunk 2]
```diff
@@ -68,11 +143,49 @@ const sign = payload => {
...
 
-const signMessage = async message => {
-  const private_key = process.env.PRIVATE_KEY;
+const signMessage = async (message, signer) => {
+  const sig = await signer.signMessage(ethers.utils.arrayify(message));
+  const { v, r, s } = ethers.utils.splitSignature(sig);
+  return { v, r, s };
+};
+
+const makeSettleMessage = (index, result) => {
```
Comment (brettvniekerk) [inline src/index.js:152]: this could also be an inferred return, unless you think its more readable this way

File: src/index.js [hunk 3]
```diff
@@ -657,6 +793,16 @@ app.get("/results/:fullid", async (req, res) => {
 
   const results = await axios(config);
 
+  if (results.status !== 200) {
+    console.error("TAB call failed", results.status);
```
Comment (brettvniekerk) [inline src/index.js:797]: this can be one line `if (results.status !== 200) return console.error(etc...)`

File: src/index.js [hunk 4]
```diff
@@ -657,6 +793,16 @@ app.get("/results/:fullid", async (req, res) => {
...
+    return;
+  }
+
+  if (results.error) {
+    console.error(results.error);
```
Comment (brettvniekerk) [inline src/index.js:802]: same here

File: src/index.js [hunk 5]
```diff
@@ -702,15 +866,86 @@ const getHistory = async placeEventFilter => {
         amount: ethers.utils.formatUnits(amount, 18),
         tx: transactionHash,
         blockNumber,
-        signature
+        result: raceResult?.toString()
```
Comment (brettvniekerk) [inline src/index.js:869]: should we not handle an `undefined` race result?

File: src/index.js [hunk 6]
```diff
@@ -702,15 +866,86 @@ const getHistory = async placeEventFilter => {
...
   }
 
   return results;
 };
 
+// Returns the result of a race for a given proposition_id
+const getRaceResult = async proposition_id => {
+  const parts = proposition_id.split("_");
+
+  // TODO: Remove this hack when we are done with the cup
```
Comment (brettvniekerk) [inline src/index.js:881]: is this supposed to be done now?

File: src/index.js [hunk 7]
```diff
@@ -702,15 +866,86 @@ const getHistory = async placeEventFilter => {
...
+  if (parts.length !== 4 && parts !== "melbournecup2022-150461") {
+    console.error({ error: "invalid proposition id" }, parts);
+    return;
+  }
+
+  const date = parts[0];
+
+  // NOTE: This only works if the date is in the past
+  // Check if the date is today, if it is, use the live results
+  // TODO: We need to not hardcode the state, perhaps we can add
```
Comment (brettvniekerk) [inline src/index.js:891]: should this not be resolved in this ticket?

File: src/index.js [hunk 8]
```diff
@@ -702,15 +866,86 @@ const getHistory = async placeEventFilter => {
...
+      // This is a bit messy, but I prefer to use strict equality
+      r => r.raceNumber === Number(parts[2])
+    );
+
+    if (result.raceStatus !== "Paying") {
+      console.error("Warning: race status is not Paying", result.raceStatus);
+    }
+
+    // Did the backed horse win?
+    // TODO: confirm result.results[0][0] will always be correct for the winner
```
Comment (brettvniekerk) [inline src/index.js:938]: should this not be done in this pr?

File: src/index.js (hunk 7 above)
Comment (OptimusOpus) [reply src/index.js:891]: No, this is to get the backing working

File: src/index.js (hunk 8 above)
Comment (OptimusOpus) [reply src/index.js:938]: No, this is to get the backing working

File: src/index.js (hunk 6 above)
Comment (OptimusOpus) [reply src/index.js:881]: I am not done with this ticket, this is to merge work in to unblock other tickets. I plan on removing that however I haven't tested it with it removed. i would rather keep it in until I have done a run through with this still there

**Review by brettvniekerk — APPROVED** [review]

**Pull request merged**